__tests__/ship.test.js

```diff
@@ -0,0 +1,70 @@
+const Ship = require("../src/ship");
+
+describe("Ship", () => {
+  describe("with ports and an itinerary", () => {
+    let ship;
+    let newcastle;
+    let amsterdam;
+    let itinerary;
+
+    beforeEach(() => {
+      newcastle = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Newcastle",
+        ships: [],
+      };
+
+      amsterdam = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Amsterdam",
+        ships: [],
+      };
+
+      itinerary = {
+        ports: [newcastle, amsterdam],
+      };
+
+      //itinerary = new Itinerary ([newcastle, amsterdam]);
+
+      ship = new Ship(itinerary);
+    });
+    it("returns an object", () => {
+      expect(ship).toBeInstanceOf(Object);
+    });
+
+    it("sets the starting port property", () => {
+      expect(ship.currentPort).toBe(newcastle);
+    });
+
+    it("gets added to port on instantiation", () => {
+      expect(newcastle.addShip).toHaveBeenCalledWith(ship);
+    });
+
+    it("can set sail from a starting port", () => {
+      ship.setSail();
+
+      expect(ship.currentPort).toBeFalsy();
+      expect(newcastle.removeShip).toHaveBeenCalledWith(ship);
+      //expect (ship.previousPort).toBe(itinerary);
```

Comment:
I think I see what you were trying to do here, it just needs a bit of a tweak. Here youre trying to assert that the whole itinerary is equal to just the first port, so its definitely going to fail. If you update it to
```
      expect (ship.previousPort).toBe(newcastle);
```
Then it will work fine